pkg/controller/jobs/statefulset/statefulset_reconciler.go

```diff
@@ -144,15 +159,53 @@ func (r *Reconciler) Update(e event.UpdateEvent) bool {
 	return r.handle(e.ObjectNew)
 }
 
-func (r *Reconciler) Delete(event.DeleteEvent) bool {
-	return false
+func (r *Reconciler) Delete(e event.DeleteEvent) bool {
+	return r.handle(e.Object)
 }
 
 func (r *Reconciler) handle(obj client.Object) bool {
 	sts, isSts := obj.(*appsv1.StatefulSet)
 	if !isSts {
-		return false
+		return true
 	}
 	// Handle only statefulset managed by kueue.
 	return jobframework.IsManagedByKueue(sts)
 }
+
+var _ handler.EventHandler = (*podHandler)(nil)
+
+type podHandler struct{}
+
+func (h *podHandler) Generic(context.Context, event.GenericEvent, workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+}
+
+func (h *podHandler) Create(_ context.Context, e event.CreateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	// To watch for pod ADDED events.
+	h.handle(e.Object, q)
+}
+
+func (h *podHandler) Update(_ context.Context, e event.UpdateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	h.handle(e.ObjectNew, q)
```

Comment:
Do we need this? What is the scenario? IIUC as long as STS exists the pod updates are reflected in the STS status updates, and so STS should reconcile.

Reply:
Yeah, I think, you're right. We don't need to handle Pod updates.